arch/RISCV/RISCVInstPrinter.c

```diff
@@ -0,0 +1,170 @@
+  	if (/*NoAliases ||*/ !printAliasInstr(NewMI, O, info))
+    		printInstruction(NewMI, O, MRI);
+  		//printAnnotation(O, Annot);
+}
+
+static void printRegName(SStream *OS, unsigned RegNo) 
+{
+  	SStream_concat0(OS, getRegisterName(RegNo, RISCV_ABIRegAltName));
+}
+
+/**
+void RISCVInstPrinter::printOperand(const MCInst *MI, unsigned OpNo,
+                                    raw_ostream &O, const char *Modifier) 
+*/
+static void printOperand(MCInst *MI, unsigned OpNo, SStream *O) 
+{
+  	unsigned reg;
+  	int64_t Imm = 0;
+
+  	MCOperand *MO = MCInst_getOperand(MI, OpNo);
+  
+  	if (MCOperand_isReg(MO)) {
+    		reg = MCOperand_getReg(MO);
+    		printRegName(O, reg);
+    		if (MI->csh->detail) {
+      			MI->flat_insn->detail->riscv.operands[MI->flat_insn->detail->riscv.op_count].type = RISCV_OP_REG;
+      			MI->flat_insn->detail->riscv.operands[MI->flat_insn->detail->riscv.op_count].reg = reg;
+      			MI->flat_insn->detail->riscv.op_count++;
+    		}
+
+  	} else {
+		assert(MCOperand_isImm(MO) && "Unknown operand kind in printOperand");
+    		Imm = MCOperand_getImm(MO);
+    		if (Imm >= 0) {
+      			if (Imm > HEX_THRESHOLD)
+        			SStream_concat(O, "0x%" PRIx64, Imm);
+      			else
+				SStream_concat(O, "%" PRIu64, Imm);
+    		} else {
+      			if (Imm < -HEX_THRESHOLD)
+				SStream_concat(O, "-0x%" PRIx64, -Imm);
+      			else
+				SStream_concat(O, "-%" PRIu64, -Imm);
+    		}
+
+    		if (MI->csh->detail) {
+      			MI->flat_insn->detail->riscv.operands[MI->flat_insn->detail->riscv.op_count].type = RISCV_OP_IMM;
+      			MI->flat_insn->detail->riscv.operands[MI->flat_insn->detail->riscv.op_count].imm = Imm;
+      			MI->flat_insn->detail->riscv.op_count++;
+    		}
+  	}
+
+  	//assert(MO.isExpr() && "Unknown operand kind in printOperand");
+	
+	return;
+}
+
+static void printCSRSystemRegister(const MCInst *MI, unsigned OpNo,
+                                   //const MCSubtargetInfo &STI,
+                                   SStream *O) 
+{
+	// TODO: Not yeat implementated.
+	return;
+  	//assert (0 && "CSR system register hav't support.");
+#if 0
+  unsigned Imm = MI->getOperand(OpNo).getImm();
+  auto SysReg = RISCVSysReg::lookupSysRegByEncoding(Imm);
+  if (SysReg && SysReg->haveRequiredFeatures(STI.getFeatureBits()))
+    O << SysReg->Name;
+  else
+    O << Imm;
+#endif
+}
+
+static void printFenceArg(MCInst *MI, unsigned OpNo, SStream *O) 
+{
+  	unsigned FenceArg = MCOperand_getImm(MCInst_getOperand(MI, OpNo));
+  	//assert (((FenceArg >> 4) == 0) && "Invalid immediate in printFenceArg");
+
+  	if ((FenceArg & RISCVFenceField_I) != 0)
+    		SStream_concat0(O, "i");
+  	if ((FenceArg & RISCVFenceField_O) != 0)
+    		SStream_concat0(O, "o");
+ 	if ((FenceArg & RISCVFenceField_R) != 0)
+    		SStream_concat0(O, "r");
+  	if ((FenceArg & RISCVFenceField_W) != 0)
+    		SStream_concat0(O, "w");
+  	if (FenceArg == 0)
+    		SStream_concat0(O, "unkonwn");
```

Comment:
this is a typo: "unknown"

Reply:
not typo, I followed the LLVM.

Reply:
then upstream should be fixed, too?

Reply:
This output will be as a mnemonic postfix, like: `W` of `FENCE.W`, looks like print "unknown" is for the purpose of future change.

Reply:
i mean you are having "unkonwn", but that is typo, because it should be "unknown".

Reply:
xDDDD